src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -17,8 +17,8 @@
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             Query<MovieSession> query = session.createQuery("select ms from MovieSession ms "
-                            + "inner join fetch ms.movie m "
-                            + "inner join fetch ms.cinemaHall "
+                            + "left join fetch ms.movie m "
+                            + "left join fetch ms.cinemaHall "
```

Comment:
In this case that doesn't matter whether you are using inner join of left. We can't have MovieSession without a movie or without cinemaHall.